mtags/src/main/scala-3/scala/meta/internal/pc/completions/AmmoniteFileCompletions.scala

```diff
@@ -0,0 +1,99 @@
+package scala.meta.internal.pc
+package completions
+
+import java.nio.file.Path
+import java.{util as ju}
+
+import scala.collection.JavaConverters.*
+
+import scala.meta.internal.mtags.MtagsEnrichments.*
+import scala.meta.internal.pc.AutoImports.AutoImport
+import scala.meta.io.AbsolutePath
+
+import dotty.tools.dotc.ast.tpd.Tree
+import dotty.tools.dotc.ast.tpd.*
+import dotty.tools.dotc.ast.untpd.ImportSelector
+import dotty.tools.dotc.core.Contexts.Context
+import dotty.tools.dotc.core.StdNames.*
+import org.eclipse.{lsp4j as l}
+
+object AmmoniteFileCompletions:
+
+  private def translateImportToPath(tree: Tree): String =
+    tree match
+      case Select(qual, name) =>
+        val pathPart = name.toString()
+        translateImportToPath(qual) + "/" + {
+          if pathPart == "^" then ".."
+          else pathPart
+        }
+      case Ident(_) =>
+        ""
+      case _ => ""
+
+  def contribute(
+      select: Tree,
+      selector: List[ImportSelector],
+      posRange: l.Range,
+      rawPath: String,
+      workspace: Option[Path],
+      rawFileName: String,
+  )(using Context): List[CompletionValue] =
+
+    val fileName = rawFileName
+      .split("/")
+      .last
+      .stripSuffix(".amm.sc.scala")
+
+    val split = rawPath
+      .split("\\$file")
+      .toList
+
+    val editRange = selector.headOption.map { sel =>
```

Comment:
I adjusted the ranges since the completions seemed to not be inserted correctly and only appended to the end.